pandas/tests/io/test_gcs.py

```diff
@@ -9,12 +9,32 @@
 from pandas.util import _test_decorators as td
 
 
-@td.skip_if_no("gcsfs")
-def test_read_csv_gcs(monkeypatch):
+@pytest.fixture
+def gcs_buffer(monkeypatch):
+    """Emulate GCS using a binary buffer."""
     from fsspec import AbstractFileSystem, registry
 
     registry.target.clear()  # noqa  # remove state
 
```

Comment:
looks fine, does this make it easier / cleaner?

Reply:
@alimcmaster1 commented one some duplication across the GCS tests. I'm not 100% sure how the monkeypatch part works, the CI seems to be happy about it. Double/triple checking that I didn't invalidated these tests would be good.